db/hybrid_logical_vector.go

```diff
@@ -122,6 +203,16 @@ type HybridLogicalVector struct {
 	PreviousVersions  map[string]string `json:"pv,omitempty"`        // map of previous versions for fast efficient lookup
 }
 
+// PersistedHLV is the version of the version vector that is persisted to the bucket, pv and mv will be lists of source version pairs in the bucket
```

Comment:
To avoid a new struct and the conversion between PersistedHLV and HybridLogicalVector, I'm wondering if we can take the same approach for PreviousVersions and MergeVersions that we do for RevTree inside SyncData - define a custom UnmarshalJSON/MarshalJSON that does the transformation from deltas to map (and vice versa).  That would properly isolate the in-bucket storage format to the persistence layer, and avoid the extra allocations/copy between PersistedHLV and HybridLogicalVector.  So basically:
- define a new type HLVVersions that's map[string]string, and use that for MergeVersions and PreviousVersions in HybridLogicalVector
- define MarshalJSON and UnmarshalJSON for HLVVersions that use PersistedDeltasToMap and VersionsToDeltas

I think this change would really help to hide the persistence details from users of HybridLogicalVector and minimize the impact of this change.